src/workerd/api/worker-rpc.c++

```diff
@@ -1167,13 +1173,13 @@ private:
     };
 
     kj::Maybe<jsg::JsValue> result;
-    kj::StringPtr methodNameForErrors;
+    kj::ConstString methodNameForTrace;
 
     switch (callParams.which()) {
       case rpc::JsRpcTarget::CallParams::METHOD_NAME: {
         kj::StringPtr methodName = callParams.getMethodName();
         result = getProperty(methodName);
-        methodNameForErrors = methodName;
+        methodNameForTrace = methodName.attach();
```

Comment:
If I'm understanding correctly, this `StringPtr::attach()` call produces a `ConstString` that does not own its own memory, but instead relies on the caller to keep `callParams` alive.

...which I assume is OK, since the only caller of `tryGetProperty()` uses the returned `methodNameForTrace` value synchronously in the same scope as the passed `params` value.

Reply:
Right.